src/components/NewDesign/Swap/Swap.tsx

```diff
@@ -0,0 +1,424 @@
+    )
+    return !!swapPool
+  }
+  const updateEstimatedAmount = (amount: string | null = null) => {
+    if (tokenFromIndex !== null && tokenToIndex !== null) {
+      setAmountTo(
+        calculateSwapOutAmount(tokens[tokenFromIndex], tokens[tokenToIndex], amount ?? amountTo)
+      )
+    }
+  }
+  const updateFromEstimatedAmount = (amount: string | null = null) => {
+    if (tokenFromIndex !== null && tokenToIndex !== null) {
+      setAmountFrom(
+        calculateSwapOutAmount(tokens[tokenToIndex], tokens[tokenFromIndex], amount ?? amountFrom)
+      )
+    }
+  }
+
+  const getButtonMessage = () => {
+    if (walletStatus !== Status.Initialized) {
+      return 'Please connect wallet'
+    }
+
+    if (tokenFromIndex === null || tokenToIndex === null) {
+      return 'Swap tokens'
+    }
+
+    if (!getIsXToY(tokens[tokenFromIndex].assetAddress, tokens[tokenToIndex].assetAddress)) {
+      return 'Pair does not exist'
+    }
+
+    if (
+      printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).eqn(0) ||
+      printBNtoBN(amountTo, tokens[tokenToIndex].decimal).eqn(0)
+    ) {
+      return 'Insufficient trade volume'
+    }
+    if (swap) {
+      if (
+        printBNtoBN(amountTo, tokens[tokenToIndex].decimal).gt(
+          tokens[tokenToIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    } else {
+      if (
+        printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).gt(
+          tokens[tokenFromIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    }
+    return 'Swap'
+  }
+
+  const setSlippage = (slippage: string): void => {
+    setSlippTolerance(slippage)
+  }
+
+  const handleClickSettings = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+    blurContent()
+    setSettings(true)
+  }
+
+  const hoverDetails = () => {
+    setDetails(!details)
+  }
+
+  const handleCloseSettings = () => {
+    unblurContent()
+    setSettings(false)
+  }
+  return (
+    <Grid container>
+      <Grid container className={classes.header}>
+        <Typography component='h1'>Swap tokens</Typography>
+        <CardMedia image={settingIcon} className={classes.settingsIcon} onClick={handleClickSettings}/>
+        <Grid className={classes.slippage}>
+          <Slippage open={settings}
+            setSlippage={setSlippage}
+            handleClose={handleCloseSettings}
+            anchorEl={anchorEl}/>
+        </Grid>
+      </Grid>
+      <Grid container className={classes.root} direction='column'>
+        <Box className={classes.tokenComponentTextContainer}>
+          <Typography className={classes.tokenComponentText}>Est.: </Typography>
+          <Typography className={classes.tokenComponentText}>
+          Balance: {tokenFromIndex !== null
+              ? swap
+                ? printBN(tokens[tokenToIndex ?? 0].balance, tokens[tokenToIndex ?? 0].decimal)
```

Comment:
why are you showing tokenTo balance in tokenFrom field?

Reply:
just use 0

Reply:
because swap animate inputs and want to show balance depends on where input is.